expression/builtin.go

```diff
@@ -656,10 +653,8 @@ type builtinFunc interface {
 	evalJSON(row []types.Datum) (val json.JSON, isNull bool, err error)
 	// getArgs returns the arguments expressions.
 	getArgs() []Expression
-	// isDeterministic checks if a function is deterministic.
-	// A function is deterministic if it returns the same result whenever it's called with the same inputs.
```

Comment:
Keep this line?

Reply:
it's difficult to illustrate whether a function can be folded.

In the past, if a function returns the same result whenever it's called with the same inputs it can be folded, but issue #4384 stated that function NOW() can be folded too, which violates the rule.

So I prefer to remove the incorrect comment